discovery/watcher.go

```diff
@@ -469,6 +471,7 @@ func (w *Watcher) switchServer(to Addr) error {
 	if err != nil {
 		return err
 	}
+	// FIXME: with the refactor, w.balancer is no longer set properly
 	return w.balancer.WaitForTransition(w.ctx, to)
```

Comment:
By moving the balancer builder registration to an `init()` function, we've lost the ability to pass a watcher reference through to wherever a balancer instance gets built internally by gRPC.

We either need to find some way to pass a watcher reference through in `grpc.DialContext`, or find some alternative way to kick off the `WaitForTransition` flow - [`resolver.ResolveNow`](https://pkg.go.dev/google.golang.org/grpc/resolver#Resolver) looks interesting, but I can't seem to figure out the right way to call it with the `resolver.ResolveNowOptions` argument.